Marlin/gcode.h

```diff
@@ -148,8 +147,7 @@ class GCodeParser {
 
     // Code is found in the string. If not found, value_ptr is unchanged.
     // This allows "if (seen('A')||seen('B'))" to use the last-found value.
-    // This is volatile because its side-effects are important
-    static volatile bool seen(const char c) {
+    static bool seen(const char& c) {
```

Comment:
@thinkyhead Like I said on the commit log, we'd like to have it volatile but gcc refuses to make it volatile so that's why it throws a warning at us "type qualifiers ignored on function return type [-Wignored-qualifiers]". I didn't investigated further why it is behaving like that and optimizing out the volatile.

`const char c` makes the compiler to make a copy the passed argument and mark it as read only inside the function. `const char& c` we're hinting the compiler that he can pass the memory address directly (this is the important part, no copy operation) of the argument because the function will not modify it.